### PR TITLE
docs(migration): update to use new `runtimeConfig`

### PR DESCRIPTION
**Pull request by tobiasdiez**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The migration guide still mentioned the deprecated `publicRuntimeConfig` and `privateRuntimeConfig`, this is changed to the new `runtimeConfig`.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/migration/8.runtime-config.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ When referencing these variables within your components, you will have to use th
 
 ## Migration
 
-1. Add any environment variables you use in your app to your `publicRuntimeConfig` or `privateRuntimeConfig`.
+1. Add any environment variables you use in your app to `runtimeConfig` in `nuxt.config.ts`.
```
Comment (DamianGlowala) [inline docs/content/migration/8.runtime-config.md:16]:
```suggestion
1. Add any environment variables that you use in your app to the `runtimeConfig` property of the `nuxt.config` file.
```

File: docs/content/migration/8.runtime-config.md [hunk 2]
```diff
@@ -25,9 +25,9 @@ import { defineNuxtConfig } from 'nuxt'
 
 export default defineNuxtConfig({
   runtimeConfig: {
-    secretKey: '', // variables that can only be accessed on server-side
+    secretKey: '', // variable that can only be accessed on server-side
```
Comment (DamianGlowala) [inline docs/content/migration/8.runtime-config.md:28]:
```suggestion
    secretKey: '', // variable that can only be accessed on the server side
```

File: docs/content/migration/8.runtime-config.md [hunk 3]
```diff
@@ -25,9 +25,9 @@ import { defineNuxtConfig } from 'nuxt'
...
     public: {
-      BASE_URL: process.env.BASE_URL || 'https://nuxtjs.org'
+      BASE_URL: process.env.BASE_URL || 'https://nuxtjs.org' // variable that can also be accessed on the client-side
```
Comment (DamianGlowala) [inline docs/content/migration/8.runtime-config.md:30]:
```suggestion
      BASE_URL: process.env.BASE_URL || 'https://nuxtjs.org' // variable that can also be accessed on the client side
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

**Comment by pi0** [comment]
Thanks @tobiasdiez @DamianGlowala

**Pull request merged**